android-java/app/src/main/java/com/example/msaldemo/SingleAccountModeFragment.java

```diff
@@ -0,0 +1,383 @@
+    Button callGraphApiSilentButton;
+    TextView scopeTextView;
+    TextView graphResourceTextView;
+    TextView logTextView;
+    TextView currentUserTextView;
+    TextView deviceModeTextView;
+
+    /* Azure AD Variables */
+    private ISingleAccountPublicClientApplication mSingleAccountApp;
+    private IAccount mAccount;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater,
+                             ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        final View view = inflater.inflate(R.layout.fragment_single_account_mode, container, false);
+        initializeUI(view);
+
+        // Creates a PublicClientApplication object with res/raw/auth_config_single_account.json
+        PublicClientApplication.createSingleAccountPublicClientApplication(getContext(),
+                R.raw.auth_config_single_account,
+                new IPublicClientApplication.ISingleAccountApplicationCreatedListener() {
+                    @Override
+                    public void onCreated(ISingleAccountPublicClientApplication application) {
+                        /**
+                         * This test app assumes that the app is only going to support one account.
+                         * This requires "account_mode" : "SINGLE" in the config json file.
+                         **/
+                        mSingleAccountApp = application;
+                        loadAccount();
+                    }
+
+                    @Override
+                    public void onError(MsalException exception) {
+                        displayError(exception);
+                    }
+                });
+
+        return view;
+    }
+
+    /**
+     * Initializes UI variables and callbacks.
+     */
+    private void initializeUI(@NonNull final View view) {
+        signInButton = view.findViewById(R.id.btn_signIn);
+        signOutButton = view.findViewById(R.id.btn_removeAccount);
+        callGraphApiInteractiveButton = view.findViewById(R.id.btn_callGraphInteractively);
+        callGraphApiSilentButton = view.findViewById(R.id.btn_callGraphSilently);
+        scopeTextView = view.findViewById(R.id.scope);
+        graphResourceTextView = view.findViewById(R.id.msgraph_url);
+        logTextView = view.findViewById(R.id.txt_log);
+        currentUserTextView = view.findViewById(R.id.current_user);
+        deviceModeTextView = view.findViewById(R.id.device_mode);
+
+        final String defaultGraphResourceUrl = MSGraphRequestWrapper.MS_GRAPH_ROOT_ENDPOINT + "v1.0/me";
+        graphResourceTextView.setText(defaultGraphResourceUrl);
+
+        signInButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                if (mSingleAccountApp == null) {
+                    return;
+                }
+
+                mSingleAccountApp.signIn(getActivity(), null, getScopes(), getAuthInteractiveCallback());
```

Comment:
Let's use the non-deprecated methods that use the SignInParameter class.

```suggestion
                final SignInParameters signInParameters = SignInParameters.builder()
                .withActivity(getActivity())
                .withLoginHint(null)
                .withScopes(getScopes)
                .withCallback(getAuthInteractiveCallback())
                .build();
                
                mSingleAccountApp.signIn(signInParameters);
```